src/task-actions.ts

```diff
@@ -106,25 +106,25 @@ export async function starknetCompileAction(args: any, hre: HardhatRuntimeEnviro
     const rootRegex = new RegExp("^" + root);
 
     const defaultSourcesPath = hre.config.paths.starknetSources;
-    const sourcesPaths: string[] = args.paths || [defaultSourcesPath];
+    const sourcesPaths: string[] = args.paths || defaultSourcesPath.split(" ");
     const artifactsPath = hre.config.paths.starknetArtifacts;
 
     let statusCode = 0;
     for (let sourcesPath of sourcesPaths) {
         if (!path.isAbsolute(sourcesPath)) {
             sourcesPath = path.normalize(path.join(root, sourcesPath));
         }
-
         checkSourceExists(sourcesPath);
         const files = await traverseFiles(sourcesPath, "*.cairo");
+        const defaultSourcesPaths = defaultSourcesPath.replace(" ", ":");
```

Comment:
What if `defaultSourcesPath` contains multiple consecutive spaces? And where is the input format specified/documented?